DirectProgramming/DPC++FPGA/Tutorials/Tools/experimental/platform_designer/add-oneapi/src/add.cpp

```diff
@@ -0,0 +1,100 @@
+#include <iostream> 
+#include <stdlib.h>
+
+// oneAPI headers
+#include <sycl/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/host_pipes.hpp>
+
+// use host pipes to write into addresses in the CSR
+using OutputPipe = sycl::ext::intel::prototype::pipe<
+    class ID_PipeOut, int, 1,
+    // choose defaults for these 4:
+    0, 1, true, false,
+    // store the most recently processed index to the CSR
+    sycl::ext::intel::prototype::internal::protocol_name::AVALON_MM>;
+
+class Add_Kernel {
+ public:
+  int a;
+  int b;
+
+  void operator()() const {
+    int sum = a + b;
+
+    OutputPipe::write(sum);
+  }
+};
+
+int main() {
+  bool passed = false;
+
+  try {
+// This design is tested with 2023.0, but also accounts for a syntax change in
+// 2023.1
+#if __INTEL_CLANG_COMPILER >= 20230100
+#if FPGA_SIMULATOR
+    std::cout << "using FPGA Simulator." << std::endl;
+    sycl::queue q(sycl::ext::intel::fpga_simulator_selector_v);
+#elif FPGA_HARDWARE
+    std::cout << "using FPGA Hardware." << std::endl;
+    sycl::queue q(sycl::ext::intel::fpga_selector_v);
+#else  // #if FPGA_EMULATOR
+    std::cout << "using FPGA Emulator." << std::endl;
+    sycl::queue q(sycl::ext::intel::fpga_emulator_selector_v);
+#endif
+#elif __INTEL_CLANG_COMPILER >= 20230000
+#if FPGA_SIMULATOR
+    std::cout << "using FPGA Simulator." << std::endl;
+    sycl::queue q(sycl::ext::intel::fpga_simulator_selector{});
+#elif FPGA_HARDWARE
+    std::cout << "using FPGA Hardware." << std::endl;
+    sycl::queue q(sycl::ext::intel::fpga_selector{});
+#else  // #if FPGA_EMULATOR
+    std::cout << "using FPGA Emulator." << std::endl;
+    sycl::queue q(sycl::ext::intel::fpga_emulator_selector{});
+#endif
+#else
+    assert(false) && "this design requires oneAPI 2023.0 or 2023.1!"
```

Comment:
You probably want to just say '2023.0 or later', rather than explicitly mentioning 2023.1, so you don't have to update this when 2023.2 comes out.

Reply:
we have to update it anyway to remove the 2023.0 selectors.